### DISCUSSION TITLE
How to use version 3.* in my old Xamarin app?

### DISCUSSION THREAD

**Discussion by andrey-val-rodin** [dotnet-bluetooth-le/dotnet-bluetooth-le#897] [Q&A]
I use NuGet package Plugin.BLE 2.2.0-pre5 in my application.

Now I want to update this package. Build succeeds, but app throws exception "This functionality is not implemented in the portable version of this assembly.  You should reference the NuGet package from your main application project in order to reference the platform-specific implementation."

**Comment by janusw** [comment]
Well, technically the 3.x versions still contain the old Xamarin targets, but I think no one has seriously cared to keep them working (since Xamarin is well out of support by now).
Is there any hard reason why you need to update to 3.x? Or, asked differently, don't you have any plans to update your app to .NET 8 (and, possibly, MAUI)? Neither the Xamarin framework itself nor any 3rd-party libs that build upon it will receive any further updates, I guess ...

**Comment by andrey-val-rodin** [comment]
No, there are no hard reasons to update the package.

In fact, exception occurs in UWP. I was able to build working version for Android after switching to target Android version 13.

I was hoping that latest version of the plugin fixed issue with thread freezing. Unfortunately, this is not the case.The problem is that if Bluetooth is turning off while Device.GetServiceAsync(ServiceUuid, token) is working, the call may never return result.
